### PR TITLE
Add documentation on how to use Managed Identity (password-less auth) with Azure Storage

### PR DESCRIPTION
**Pull request by tonybaloney**
Managed Identity is the recommended approach for authenticating to Azure Storage Accounts from a Python app running on Azure (or when debugging in VS Code).

This extension already supports it, this PR gives an example of how to do it.

### CODE REVIEW

File: docs/backends/azure.rst [hunk 1]
```diff
@@ -46,11 +46,53 @@ Authentication Settings
...
+#. ``token_credential`` or ``AZURE_TOKEN_CREDENTIAL`` with ``account_name`` or ``AZURE_ACCOUNT_NAME``
 #. ``sas_token`` or ``AZURE_SAS_TOKEN``
 
+Using Managed Identity
+++++++++++++++++++++++
+
+`Azure Managed Identity <https://learn.microsoft.com/entra/identity/managed-identities-azure-resources/overview>`_ is an authentication method that allows you to authenticate to Azure services without storing credentials in your code.
+Managed Identity is the recommended mechanism for password-less authentication to Azure Storage Accounts from other Azure services like App Services, Functions, Container Apps, and VMs.
+
+To use Managed Identity you will need to configure a System Assigned Managed Identity or a User Assigned Managed Identity for your app service. Then you can use the ``DefaultAzureCredential`` class from the Azure SDK to authenticate.
```
Comment (pamelafox) [inline docs/backends/azure.rst:60]: Will it work with user-assigned? I dont see a way to specify the client ID of a user-assigned identity?

Comment (tonybaloney) [reply docs/backends/azure.rst:60]: Yes, you would pass the parameter to the `DefaultAzureCredential` call. e.g. `DefaultAzureCredential(managed_identity_client_id=client_id)`

**Review by pamelafox — APPROVED** [review]

**Comment by jschneier** [comment]
Hi, I just merged a different Azure docs update, can you please rebase and I will then merge.

**Comment by tonybaloney** [comment]
@jschneier done!

**Pull request merged**